### PR TITLE
#26 Allow user to embed videos to a post using a google drive url

### PR DESCRIPTION
**Pull request by naomi-eliasar**
This PR solves #26 
It makes it possible to add a google drive video with the google drive share link.

A new field is added to the blog-post form to add a google drive share link for a video.
On the blog page this video is embedded below the body and above the image gallery

### CODE REVIEW

File: src/blog/migrations/0006_authorindexpage_blogpage_author_authorpage.py [hunk 1]
```diff
@@ -10,7 +10,6 @@ class Migration(migrations.Migration):
 
     dependencies = [
         ('blog', '0005_alter_blogpage_date'),
-        ('wagtailcore', '0093_uploadedfile'),
```
Comment (drikusroor) [inline src/blog/migrations/0006_authorindexpage_blogpage_author_authorpage.py:13]: Why did you remove this? It's generally a bad sign if you change migrations afterwards 🤔

File: src/blog/migrations/0008_alter_blogpage_video.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        ('blog', '0007_blogpage_video'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='blogpage',
+            name='video',
+            field=models.URLField(blank=True, null=True),
+        ),
+    ]
```
Comment (drikusroor) [inline src/blog/migrations/0008_alter_blogpage_video.py:18]: To make it a bit cleaner, I would merge this migration (0008) and the previous one (0007) into one migration. You can do this in two ways:

## Method 1. (re-do migrations)

```sh
# Revert the last two migrations (go back to 0006)
bin/manage migrate blog 0006

# Delete the last two migrations
rm src/blog/migrations/0008_alter_blogpage_video.py
rm src/blog/migrations/0007_blogpage_video.py

# Re-create migrations
bin/manage makemigrations

# Apply newly created blog migrations
bin/manage migrate blog
```

## Method 2. (squash migrations)

See also: https://docs.djangoproject.com/en/dev/topics/migrations/#squashing-migrations

```sh
# squash the last two migrations with squasmigrations <app> <from> <to>
bin/manage squashmigrations blog 0007 0008
```

Then, after we are sure that the migrations have been applied on the server as well, we can remove the old migrations 0007 & 0008

File: src/blog/models.py [hunk 3]
```diff
@@ -90,6 +90,8 @@ class BlogPage(Page):
     )
     intro = models.CharField(max_length=250)
     body = RichTextField(blank=True)
+    video = models.URLField(null=True, blank=True, help_text="Google drive share link",
+)
```
Comment (drikusroor) [inline src/blog/models.py:94]: I guess this is fine for now, especially if we are sure we only want to upload one video per post. But we can also think about creating custom blocks: https://docs.wagtail.org/en/stable/advanced_topics/customisation/streamfield_blocks.html

File: src/blog/templates/blog/blog_page.html [hunk 4]
```diff
@@ -38,20 +40,41 @@ <h1 class="bold text-4xl mb-0 text-black">{{ page.title }}</h1>
...
+                    allowFullScreen
+                    loading="lazy"
+                ></iframe>
+            </div>
+
+            <!-- Image Gallery -->
             <div class="p-4 grid md:grid-cols-2 lg:grid-cols-3 gap-4">
                 {% for item in page.gallery_images.all %}
-                    <div class="bg-white rounded-xl overflow-hidden drop-shadow group hover:rotate-3 transition-transform">
+                    <div class="bg-white rounded-xl overflow-hidden drop-shadow group hover:rotate-3 transition-transform" data-url="{{page.video}}">
```
Comment (drikusroor) [inline src/blog/templates/blog/blog_page.html:65]: Should this `data-url` attribute still be there?

**Review by drikusroor — CHANGES_REQUESTED** [review]
It would be cleaner if you can turn the two migrations into one migration. But I'm especially curious as to why you've edited an existing migration.

File: src/blog/migrations/0006_authorindexpage_blogpage_author_authorpage.py (hunk 1 above)
Comment (naomi-eliasar) [reply src/blog/migrations/0006_authorindexpage_blogpage_author_authorpage.py:13]: It wouldn't migrate. It gave an error that these two dependencies did not exist 😅

Comment (drikusroor) [reply src/blog/migrations/0006_authorindexpage_blogpage_author_authorpage.py:13]: Let me see if I can re-produce it before we merge it. I'll get back on it

File: src/blog/migrations/0008_alter_blogpage_video.py (hunk 2 above)
Comment (naomi-eliasar) [reply src/blog/migrations/0008_alter_blogpage_video.py:18]: I squashed 0007 and 0008

File: src/blog/models.py (hunk 3 above)
Comment (naomi-eliasar) [reply src/blog/models.py:90]: As of now, I only see myself posting 1 vlog/video per post. So I think it should be fine
But maybe we can do the custom block as a refactor/upgrade later on if we want it
